### PR TITLE
ISPN-14648 GETDEL Command

### PR DESCRIPTION
**Pull request by rigazilla**
https://issues.redhat.com/browse/ISPN-14648

### CODE REVIEW

File: server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java [hunk 1]
```diff
@@ -451,4 +451,30 @@ protected RedisPubSubCommands<String, String> createPubSubConnection() {
...
+   }
+
+   @Test
+   public void testAppendNotPresent() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      String key = "append";
+      String app = " World";
+      String expect = app;
+      long retVal = redis.append(key, app);
+      assertEquals(retVal, expect.length());
```
Comment (karesti) [inline server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java:475]: assertj

**Review by karesti — CHANGES_REQUESTED** [review]

**Comment by rigazilla** [comment]
based on #10876  for assertj

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   public APPEND() {
+      super(3, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      if (arguments.size() != getArity()-1) {
+         ByteBufferUtils.stringToByteBuf("-ERR wrong number of arguments for 'append' command\r\n", handler.allocatorToUse());
```
Comment (tristantarrant) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java:28]: I would centralize all `-ERR` strings into a `Messages` class using a `MessageBundle` (see the CLI).

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+public class APPEND extends RespCommand implements Resp3Command {
+   public APPEND() {
+      super(3, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      if (arguments.size() != getArity()-1) {
```
Comment (tristantarrant) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java:27]: formatting seems a bit off here (spaces around the subtraction)

Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java]: fixed in the APPEND PR, this depends on #10876 for assertj

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java (hunk 2 above)
Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java]: sounds good, but I would trace it in a separate issue

Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/string/APPEND.java]: created https://issues.redhat.com/browse/ISPN-14840

File: server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java [hunk 4]
```diff
@@ -88,8 +89,7 @@ public String getName() {
       // DEL should always be first here
       indexedRespCommand[3] = new RespCommand[]{new DEL(), new DECR(), new DECRBY()};
       indexedRespCommand[4] = new RespCommand[]{new ECHO()};
-      // GET should always be first here
```
Comment (wburns) [inline server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java:91]: Don't delete this comment :)

**Review by wburns — APPROVED** [review]
LGTM, just minor change

File: server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java (hunk 4 above)
Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java:91]: ouch :slightly_smiling_face: fixed

**Comment by rigazilla** [comment]
rebased, let's see...

**Review by karesti — APPROVED** [review]

**Pull request merged**